### PR TITLE
fix(ui): Fix warning for nested <a> elements

### PR DESCRIPTION
**Pull request by leedongwei**
```
Warning: validateDOMNesting(...): <a> cannot appear as a descendant of <a>.
    in a (at crashHeader.jsx:109)
    in InnerReference (created by Context.Consumer)
```

This error is caused by `<EventDataSection/>` wrapping the title of each section with a `<Permalink/>`, which is implemented with `<a>`. 

Switched to a `<span>` to fix it, and because it's still wrapped by the permalink's anchor, the cursor is still a pointer on hover.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/eventDataSection.tsx [hunk 1]
```diff
@@ -159,7 +159,7 @@ const SectionHeader = styled('div')<{isCentered?: boolean}>`
 
     text-transform: none;
   }
-  & small a {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/events/eventDataSection.tsx:162]: Is `CrashHeader` the only thing that uses this css?

**Review by billyvg — COMMENTED** [review]
Does this negatively impact the position of the GuideAnchor?

Damn, `SectionHeader` has a lot of nested css -- I think this is a good set of minimal changes to fix this warning if crashHeader is the only thing that is affected

File: src/sentry/static/sentry/app/components/events/eventDataSection.tsx (hunk 1 above)
Comment (leedongwei) [reply src/sentry/static/sentry/app/components/events/eventDataSection.tsx:162]: Yep. I did a search on `<small>` in `src/sentry/static/sentry/app/components/events/*`.

**Comment by leedongwei** [comment]
Fix-up for failed snapshot.

**Comment by leedongwei** [comment]
@billyvg I'm waiting for Sercy/Blursy to check if there's a minuscule diff in `GuideAnchor`.

**Review by priscilawebdev — APPROVED** [review]
Thanks for fixing it!! 👏🙃  it seems there is no snap diff

**Review by markstory — APPROVED** [review]

**Review by billyvg — APPROVED** [review]

**Pull request merged**